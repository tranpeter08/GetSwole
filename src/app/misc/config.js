import 'dotenv';

export const API_BASE_URL = 'https://get-swole-api.herokuapp.com';
