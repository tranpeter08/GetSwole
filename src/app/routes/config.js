import Landing from '../landing/components/Landing';
import Login from '../user/containers/Login';
import Register from '../user/containers/Register';
import UserPage from '../user/containers/UserPage';
import NotAuthorized from '../auth/container/NotAuthorized';

export const config = [
  {path: '/login', component: Login},
  {path: '/register', component: Register},
  {path: '/user/:username', component: UserPage},
  {path: '/unauthorized', component: NotAuthorized},
  {path: '/', component: Landing}
];