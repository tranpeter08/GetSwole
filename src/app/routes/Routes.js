import React from 'react';
import {Route, withRouter, Switch} from 'react-router-dom';
import Navigation from './navigation/containers/Navigation';
import Landing from './landing/components/Landing';
import Login from './user/containers/Login';
import Register from './user/containers/Register';
import UserPage from './user/containers/UserPage';
import RefreshModal from './RefreshModal';
import NotAuthorized from './auth/container/NotAuthorized';
import NotFound from './misc/components/NotFound';
import config from './config';

function Routes() {
  const routes = config.map(({path, location,component}) => (
    <Route path={path} location={location} component={component}/>
  ));

  return (
    <Switch>
      {routes}
    </ Switch>
  );
}