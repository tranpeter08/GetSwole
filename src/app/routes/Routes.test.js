import React from 'react';
import {MemoryRouter} from 'react-router-dom';
import {shallow, mount} from 'enzyme';
import Routes from './Routes';
import {config} from './config';

describe('<Routes />', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<Routes />);

    expect(wrapper).toMatchSnapshot();
  });

  it('contains the correct number of routes', () => {
    const wrapper = shallow(<Routes />);
    expect(wrapper.children()).toHaveLength(config.length);
  });
})