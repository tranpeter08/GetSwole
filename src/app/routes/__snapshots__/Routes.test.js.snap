// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Routes /> renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Routes />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Route
          component={[Function]}
          path="/login"
        />,
        <Route
          component={[Function]}
          path="/register"
        />,
        <Route
          component={[Function]}
          path="/user/:username"
        />,
        <Route
          component={[Function]}
          path="/unauthorized"
        />,
        <Route
          component={[Function]}
          path="/"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "/login",
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/login",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "/register",
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/register",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "/user/:username",
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/user/:username",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "/unauthorized",
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/unauthorized",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "/",
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            path="/login"
          />,
          <Route
            component={[Function]}
            path="/register"
          />,
          <Route
            component={[Function]}
            path="/user/:username"
          />,
          <Route
            component={[Function]}
            path="/unauthorized"
          />,
          <Route
            component={[Function]}
            path="/"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "/login",
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/login",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "/register",
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/register",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "/user/:username",
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/user/:username",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "/unauthorized",
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/unauthorized",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "/",
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
